### DISCUSSION TITLE
receptionStartedSignal for LayeredEthernetInterface missing

### DISCUSSION THREAD

**Discussion by haug-den-lucas** [inet-framework/inet#978] [General]
I have a scenario, with two TsnDevices (one sender and one receiver), where I need to keep track and perform actions based on the following four timestamps

![Untitled Diagram](https://github.com/inet-framework/inet/assets/1706621/2e25430b-9aa9-469b-9902-ffc45f2854ed)

To my understanding, in INET this corresponds to:

- t1: transmissionStartedSignal
- t2: transmissionEndedSignal
- t3: receptionStartedSignal
- t4: receptionEndedSignal

I think t1, t2 and t4 work just as they should and are emitted by the PacketTransmitter and PacketReceiver in the phyLayer of LayeredEthernetInterface.
However, it seems to me, that the receptionStartedSignal is not implemented here at all.
Also to me it does not seem like there is a simple one-line fix for this problem.

Would be great if this could be fixed, or if someone can guide me in the right direction so I can fix it myself.

**Comment by levy** [comment]
Try using EthernetStreamingPhyLayer instead of EthernetPhyLayer. The default implementation doesn't provide that signal but the streaming variant should. Let me know if it works for you!

**Reply by haug-den-lucas** [reply]
Thanks for the fast response.
The receptionStartedSignal seems to be emitted as intended now.
However, if i try to add a tag to the packet using addTagIfAbsent, that tag is present when entering the DestreamingReceiver, but is gone when it leaves the DestreamingReceiver

**Reply by levy** [reply]
That's because the streaming operations use a different packet instance for start/end. This limitation is enforced by the API, because there are two distinct calls each passing the ownership of the packet to the callee, so it can't be the same packet. You can add a tag to the content (chunk) of the packet instead, because that remains the same.
